### PR TITLE
MongoChemServer changes to query spectra 

### PR DESCRIPTION
**Pull request by muammar**
This PR intends to close https://github.com/OpenChemistry/42/issues/7. It contains the following changes:

1. A new `nist` module was added. It contains functions to download `jdx` files from NIST. It adds two dependecies: `bs4`, and `jcamp`.
1.  A `find_experiment()` function was modified to operate with the new added `nist` module. It returns a dictionary with spectrum's data. 

TODO:

- [x]  Make the API accept `params['inchi']` for inchi strings instead of using `params['molecularFormula']`.  to pass the inchi string. Bad hack but worked for proof of concept.

### CODE REVIEW

File: girder/molecules/server/nist.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import re
+import requests
+from bs4 import BeautifulSoup
```
Comment (cjh1) [inline girder/molecules/server/nist.py]: We will need to add these to the [requirements.txt](https://github.com/OpenChemistry/mongochemserver/blob/master/requirements.txt)

File: girder/molecules/server/experiment.py [hunk 2]
```diff
@@ -80,15 +81,29 @@ def create(self, body):
...
-        query = { }
-        if 'molecularFormula' in params:
-            query['molecularFormula'] = params['molecularFormula']
+        if 'source' in params.keys():
+            from .nist import search_nist_inchi, get_jdx
+            from jcamp import jcamp_read
+
+            inchi = params['molecularFormula']      # TODO change API to accept
+                                                    # a different key. From
+                                                    # molecularFormula to inchi?
```
Comment (cjh1) [inline girder/molecules/server/experiment.py]: Yes, lets allow inchi to be passed in it own parameter.

Comment (muammar) [reply girder/molecules/server/experiment.py]: I was checking that yesterday, but have not found the way of doing it. I will keep lurking into the code to understand better.

File: girder/molecules/server/nist.py (hunk 1 above)
Comment (muammar) [reply girder/molecules/server/nist.py]: Done https://github.com/OpenChemistry/mongochemserver/pull/96/commits/ccc9310792ab957fd414bf3615d4991ee2f34f9a.

File: girder/molecules/server/experiment.py (hunk 2 above)
Comment (cjh1) [reply girder/molecules/server/experiment.py]: You should just be able to add it as a query parameter when you make your request [here](https://github.com/muammar/openchemistrypy/blob/ir_spec/openchemistry/__init__.py#L790)

Comment (muammar) [reply girder/molecules/server/experiment.py]: I tried doing that but the API threw an error that the key did not exist. I will try to reproduce it to post the traceback here.

Comment (cjh1) [reply girder/molecules/server/experiment.py]: You may have to add a check to see which key exists.

Comment (cjh1) [reply girder/molecules/server/experiment.py]: Feel free to push a WIP commit and I can see what is going on.

Comment (muammar) [reply girder/molecules/server/experiment.py]: Oh, I see now. I misspelled something. Now it works.  This is done, too now.

**Comment by cjh1** [comment]
@muammar Can you remove the WIP from you commit, now that you have thinks fixed up.

File: girder/molecules/server/experiment.py [hunk 3]
```diff
@@ -80,15 +81,27 @@ def create(self, body):
     def find_experiment(self, params):
         user = getCurrentUser()
 
-        query = { }
-        if 'molecularFormula' in params:
-            query['molecularFormula'] = params['molecularFormula']
+        if 'source' in params.keys():
```
Comment (cjh1) [inline girder/molecules/server/experiment.py]: `'source' in params` is enough. Actually where is `source` used?

Comment (muammar) [reply girder/molecules/server/experiment.py]: I put it there because the function can be extended to use later other sources different from NIST.

**Comment by muammar** [comment]
@cjh1 removed WIP.

File: girder/molecules/server/experiment.py (hunk 3 above)
Comment (muammar) [reply girder/molecules/server/experiment.py]: I plan to let users decide the `source` to query the spectra.

**Pull request merged**